pkg/apis/postgres-operator.crunchydata.com/v1beta1/crunchy_bridgecluster_types.go

```diff
@@ -79,70 +87,92 @@ type CrunchyBridgeClusterSpec struct {
 	Storage resource.Quantity `json:"storage"`
 }
 
+type CrunchyBridgeClusterRoleSpec struct {
+	// Name of the role within Crunchy Bridge.
+	// More info: https://docs.crunchybridge.com/concepts/users
+	Name string `json:"name"`
+
+	// The name of the Secret that will hold the role credentials.
+	// +kubebuilder:validation:Pattern=`^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`
+	// +kubebuilder:validation:Type=string
+	SecretName string `json:"secretName"`
```

Comment:
I think there was some discussion around why we want to let them name the secret -- maybe something to do with the role names from Bridge? If so, might be worth a code comment.

Reply:
OK, from chatting with Drew: we can't use role as the secret name because of disallowed characters/different constraints around role names and secret names, so we want to define the secret on the pgo side; and rather than define our secret pattern in PGO, we will let the users define their own roles. 

(I am commenting here to keep this comment _out_ of the CRD.)

(And as discussed with Drew, letting users define secret names means we need to have some guards against duplicate names within the namespace (whether those secret names come from one or multiple clusters).)